src/VTK/VtkIjkGridRepresentation.cxx

```diff
@@ -96,12 +96,10 @@ vtkSmartPointer<vtkPoints> VtkIjkGridRepresentation::createpoint()
 		jCellCount = ijkGridRepresentation->getJCellCount();
 		kCellCount = ijkGridRepresentation->getKCellCount();
 
-
-#ifdef PARAVIEW_USE_MPI
 		const ULONG64 kInterfaceNodeCount = ijkGridRepresentation->getXyzPointCountOfKInterfaceOfPatch(0);
 
 		checkHyperslabingCapacity(ijkGridRepresentation);
-		if (isHyperslabed) {
+		if (isHyperslabed && !ijkGridRepresentation->isNodeGeometryCompressed()) {
 			initKIndex = getIdProc() * (kCellCount / getMaxProc());

```

Comment:
Just a detail for optimization.
optim = kCellCount / getMaxProc()
initKIndex = getIdProc() * optim 
maxKIndex = initKIndex + optim